internal/gcsx/prefix_bucket_test.go

```diff
@@ -514,3 +514,46 @@ func TestCreateFolder(t *testing.T) {
 	_, err = bucket.GetFolder(ctx, suffix)
 	assert.NoError(t, err)
 }
+
+func TestMoveObject(t *testing.T) {
+	var notFoundErr *gcs.NotFoundError
+	prefix := "foo_"
+	var err error
+	suffix := "test"
+	wrapped := fake.NewFakeBucket(timeutil.RealClock(), "some_bucket", gcs.Hierarchical)
+	bucket, err := gcsx.NewPrefixBucket(prefix, wrapped)
+	require.NoError(t, err)
+	ctx := context.Background()
+	contents := "foobar"
+
+	// Create an object through the back door.
+	_, err = storageutil.CreateObject(ctx, bucket, suffix, []byte(contents))
+	AssertEq(nil, err)
+
+	// Move it to a new name.
+	newSuffix := "burrito"
+	o, err := bucket.MoveObject(
+		ctx,
+		&gcs.MoveObjectRequest{
+			SrcName: suffix,
+			DstName: newSuffix,
+		})
+
+	assert.NoError(t, err)
+	assert.Equal(t, newSuffix, o.Name)
+
+	// Read it through the back door.
+	actual, err := storageutil.ReadObject(ctx, bucket, o.Name)
```

Comment:
It was working because I was using a prefix bucket instead of fake bucket when calling the ReadObject and CreateObject methods. This caused the prefix bucket implementation to be invoked, adding the prefix accordingly.